tests/RobotFramework/tests/tedge_agent/http_server/entity_store_api.robot

```diff
@@ -0,0 +1,72 @@
+*** Settings ***
+Resource            ../../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Test Teardown       Get Logs    ${DEVICE_SN}
+
+Test Tags           theme:tedge_agent
+
+
+*** Variables ***
+${DEVICE_SN}    ${EMPTY}    # Main device serial number
+
+
+*** Test Cases ***
+CRUD apis
+    Execute Command
+    ...    curl -X POST http://localhost:8000/tedge/entity-store/v1/entities -H 'Content-Type: application/json' -d '{"@topic-id": "device/child01//", "@type": "child-device"}'
+
+    ${get}=    Execute Command    curl http://localhost:8000/tedge/entity-store/v1/entities/device/child01//
+    Should Be Equal
+    ...    ${get}
+    ...    {"@topic-id":"device/child01//","@parent":"device/main//","@type":"child-device"}
+    Should Have MQTT Messages
+    ...    te/device/child01//
+    ...    message_contains="@type":"child-device"
+
+    Execute Command
+    ...    curl -X PUT http://localhost:8000/tedge/entity-store/v1/entities -H 'Content-Type: application/json' -d '{"@topic-id": "device/child01//", "@type": "child-device", "@id": "child01"}'
```

Comment:
Shouldn't updating an entity item use the path to the entity?

```
curl -X PUT http://localhost:8000/tedge/entity-store/v1/entities/device/child01// -H 'Content-Type: application/json' -d '{"@topic-id": "device/child01//", "@type": "child-device", "@id": "child01"}'
```

As the `HTTP GET` is using the entity path, e.g. `curl http://localhost:8000/tedge/entity-store/v1/entities/device/child01//`

Reply:
I was a bit confused here as to what the right convention is. Since this API is more of an `upsert` rather than an `update`, I thought it's better to keep the API similar to that of `create`(`POST`). If it was a pure `update` (`PATCH`), the topic id definitely would have been accepted via the path as per the conventions.

Reply:
The convention is that the URL is the identifier of the resource to create, read, update or delete.

```
curl -X PUT http://localhost:8000/tedge/entity-store/v1/entities/device/child01// \
       -H 'Content-Type: application/json' \
      -d '{"@type": "child-device", "@id": "child01"}'
```

Reply:
> The convention is that the URL is the identifier of the resource to create, read, update or delete.

Yeah, I also would have liked that consistency in the paths. But when we discussed this last time, esp for the `POST` request, I was informed that the convention for `POST` requests to create a new resource is typically accepted to the path of the collection (`/entity-store/v1/entities` in this case) rather than the resource identifier itself.

The difference in our case, compared to a conventional case is that the unique identifier for the resource is provided by the user rather than the system. So, may be it's okay to go against the above mentioned "generic convention" in our case and accept the topic id in the URL itself for all CRUD operations.

Reply:
Yes the `POST` call is fine to use the `/entities` endpoint, but the `PUT`, `DELETE`, `GET` should use the resource name, e.g. `entities/device/child01//`

Reply:
The main advantage of having the `POST` not requiring the `/entities/...` path is that on some systems you may not actually know the resource path to an already created item, so creating the resource via the `/entities` endpoint allows the API to then return a reference to the resource url related to the created entity.

Reply:
> See Define API operations in terms of HTTP methods

The example shows that the `POST` method does not use a URL pointing to the resource, it only points to the type of resource being created (in their example it is `/customers` instead of our case of `/entities`)

Reply:
Reverted the PUT implementation as discussed offline. A proper implementation as suggested here would be introduced in a follow-up PR.